src/hd.rs

```diff
@@ -21,33 +21,31 @@ lazy_static! {
 
 #[derive(Debug)]
 pub struct HDWatcher {
```

Comment:
The structs here should probably be renamed to something like `WalletManager` and `Wallet`. But we can leave that to the end, to keep the diffs more manageable for now.